extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java

```diff
@@ -148,11 +148,12 @@ public void openshiftRequirementsJvm(OpenshiftConfig openshiftConfig,
             });
             //In all other cases its the responsibility of the image to set those up correctly.
             if (!baseImage.isPresent()) {
-                List<String> args = new ArrayList<>();
-                args.addAll(config.jvmArguments);
-                args.addAll(Arrays.asList("-jar", pathToJar));
+                List<String> cmd = new ArrayList<>();
+                cmd.add("java");
+                cmd.addAll(config.jvmArguments);
+                cmd.addAll(Arrays.asList("-jar", pathToJar));
                 envProducer.produce(KubernetesEnvBuildItem.createSimpleVar("JAVA_APP_JAR", pathToJar, null));
-                commandProducer.produce(new KubernetesCommandBuildItem("java", args.toArray(new String[args.size()])));
+                commandProducer.produce(new KubernetesCommandBuildItem(cmd.toArray(new String[cmd.size()])));
```

Comment:
```suggestion
                commandProducer.produce(new KubernetesCommandBuildItem(cmd));
```

Reply:
KubernetesCommandBuildItem requires an Array, not a List.

Reply:
If the command is converted back to a list internally, why not take a List instead?

Reply:
With array, we have the ability to write commands like ``KubernetesCommandBuildItem(new String[] { "java", "-jar", "jar.jar" })`("java")` or `KubernetesCommandBuildItem("java", "-jar", "jar.jar")` or `KubernetesCommandBuildItem(new String[] { "java", "-jar", "jar.jar" })`. This is how it was originally designed and didn't want to change as it would require to change other parts of the code as well.

We could certainly add a new constructor to accept a list and internally keeping the list instead of an array, but as I said, it would require to change the consumers of the KubernetesCommandBuildItem.

Reply:
I think it all depends on how it is used right now, really. The existing code seems to use a list for the arguments, so using an array might seem unnecessary here.

Reply:
In other parts, we're using the arrays: https://github.com/quarkusio/quarkus/pull/17889/files#diff-c15bcfcc919c2e97bb2cb66a3a836eb0537fbfb3a3fcfbd3373ad5df09749b63R206

Reply:
I've introduced a few factory methods in KubernetesCommandBuildItem to use either a command of array, or a list, or a combination of a command and arguments.
Let me know how looks like now.